### PR TITLE
File kasan crash with address range

### PR DESCRIPTION
**Pull request by kennethkang**
Rev 1:
* make `c\d+\s+\d+\s` optional for `_filter_log`
* add observed kernel crash to a test case
* add new match for kernel crash with address range
* add unit tests

### CODE REVIEW

File: src/clusterfuzz/stacktraces/__init__.py [hunk 1]
```diff
@@ -759,15 +759,29 @@ def parse(self, stacktrace: str) -> CrashInfo:
             type_filter=get_fault_description_for_android_kernel)
 
       # Generic KASan errors.
-      if self.update_state_on_match(
+      address_match = self.update_state_on_match(
```
Comment (oliverchang) [inline src/clusterfuzz/stacktraces/__init__.py:762]: Nit: no need for these variables. Follow the existing style, adding comments where necessary.

File: src/clusterfuzz/stacktraces/__init__.py [hunk 2]
```diff
@@ -1131,6 +1145,18 @@ def parse(self, stacktrace: str) -> CrashInfo:
           state.crash_address = '0x%s' % android_kernel_match.group(1)
         continue
 
+      # Android kernel stack frame without address
+      android_kernel_match = self.add_frame_on_match(
```
Comment (oliverchang) [inline src/clusterfuzz/stacktraces/__init__.py:1149]: same here.

**Review by oliverchang — CHANGES_REQUESTED** [review]

File: src/clusterfuzz/stacktraces/__init__.py (hunk 1 above)
Comment (kennethkang) [reply src/clusterfuzz/stacktraces/__init__.py:762]: 1. first and second cases do slightly different things: namely the first adds `0x` to the crash_address. Second case does not.
2. When I was following the existing style, linter kept it quite ugly with 10+ lines crammed in `if()` statement. Not easily readable.

Comment (oliverchang) [reply src/clusterfuzz/stacktraces/__init__.py:762]: Re 1) Yes they are two cases so I'm suggesting that we have two ifs without these vars. 

Re 2) It's much better for overall readability to be consistent here with the rest of the code in this function/file.

Comment (kennethkang) [reply src/clusterfuzz/stacktraces/__init__.py:762]: Oh I get what you mean. Thanks for clarifying

File: src/clusterfuzz/stacktraces/__init__.py (hunk 2 above)
Comment (kennethkang) [reply src/clusterfuzz/stacktraces/__init__.py]: fixed

File: src/clusterfuzz/stacktraces/__init__.py (hunk 1 above)
Comment (kennethkang) [reply src/clusterfuzz/stacktraces/__init__.py]: fixed

**Comment by kennethkang** [comment]
Rev 2:
* Remove unnecessary variables

**Review by oliverchang — APPROVED** [review]

**Review by oliverchang — APPROVED** [review]

**Pull request merged**